File: src/main/java/org/dxworks/githubminer/dto/export/PrBranchDTO.java
package org.dxworks.githubminer.dto.export;

import com.google.api.client.util.Key;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.dxworks.githubminer.dto.response.repository.pullrequests.PullRequestBranch;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class PrBranchDTO {
    @Key
    private String commit;
    @Key
    private String label;
    @Key
    private String ref;
    @Key
    private UserDTO user;
    @Key
    private RepoDTO repo;

    public static PrBranchDTO fromPullRequestBranch(PullRequestBranch branch) {
        if (branch == null || com.google.api.client.util.Data.isNull(branch))
            return null;
        return builder()
                .commit(branch.getSha())
                .label(branch.getLabel())
                .ref(branch.getRef())
                .user(UserDTO.fromUser(branch.getUser()))
                .repo(RepoDTO.fromRepository(branch.getRepo()))
                .build();
    }
}

File: src/main/kotlin/org/dxworks/githubminer/dto/export/PrBranchDTO.kt
package org.dxworks.githubminer.dto.export

import com.google.api.client.util.Data
import com.google.api.client.util.Key
import org.dxworks.githubminer.dto.response.repository.pullrequests.PullRequestBranch

class PrBranchDTO {
    @Key
    var commit: String? = null

    @Key
    var label: String? = null

    @Key
    var ref: String? = null

    @Key
    var user: UserDTO? = null

    @Key
    var repo: RepoDTO? = null

    companion object {
        fun fromPullRequestBranch(branch: PullRequestBranch?): PrBranchDTO? {
            return if (branch == null || Data.isNull(branch)) null else PrBranchDTO().apply {
                    commit = branch.sha
                    label = branch.label
                    ref = branch.ref
                    user = UserDTO.fromUser(branch.user)
                    repo = RepoDTO.fromRepository(branch.repo)
            }
        }
    }
}
